src/storm/mesos/logviewer/LogViewerController.java

```diff
@@ -0,0 +1,210 @@
+       
+    }
+    
+    public void setUrlDetector(IUrlDetection urlDetector) {
+        this.urlDetector = urlDetector;
+    }
+    
+    public IUrlDetection getUrlDetector() {
+        return urlDetector;
+    }
+    
+    public void setInstallLocation(String location) {
+        installLocation = location;
+    }
+    
+    public String getInstallLocation() {
+        return installLocation;
+    }
+    
+    protected Process process;
+    protected IUrlDetection urlDetector;
+    protected String installLocation;
+    protected Map<String,Object> config;
+    protected Optional<Integer> port;
+    protected static final String LOG_VIEWER_PORT = "logviewer.port";
+       
+    protected void setProcess(Process process) {
+        this.process = process;
+    }
+    
+    protected Process getProcess() {
+        return process;
+    }
+    
+    protected void setPort(Optional<Integer> oport) {
+        port = oport;
+    }
+    
+    protected Optional<Integer> getPort() {
+        return port;
+    }
+    
+    protected void setConfig(Map<String,Object> configuration) {
+        config = configuration;
+    }
+    
+    protected Map<String,Object> getConfig(){
+        return config;
+    }
+    
+    protected void launchLogViewer () throws IOException {
+         
+        if(LOG.isDebugEnabled())
+            LOG.debug("launchLogViewer: home dir? " + getInstallLocation());
+        
+        ProcessBuilder pb = createProcessBuilder(getInstallLocation());
+        setProcess(pb.start());       
+    }
+    
+    /**
+     * Create a process builder to launch the log viewer
+     * @param logDirectory
+     * @return
+     */
+    protected ProcessBuilder createProcessBuilder(String homeDirectory) {
+        ProcessBuilder pb = new ProcessBuilder(homeDirectory + "/bin/storm","logviewer");
+         
+        // If anything goes wrong at startup we want to see it.
+        File log = new File(homeDirectory + "/logs/logviewer-startup.log");
+        pb.redirectErrorStream(true);
+        pb.redirectOutput(Redirect.appendTo(log)); 
+        return pb;
+    }
+    
+    @SuppressWarnings("unchecked")
+    protected void initialize() {
+        setInstallLocation(new File("./").getAbsolutePath());
+        
+        loadConfig();
+        
+        // If the storm configuration has defined a port use it.
+        if (getPort().isPresent()) {
+            getUrlDetector().setPort(getPort());
+        }
+        else {
+            // Otherwise use the default 
+            LOG.info("No valid port in storm.yaml, using the on configured in the Url Detector " + 
+                getUrlDetector().getPort());
+            setPort(getUrlDetector().getPort());
+        }        
+    }
+    
+    @SuppressWarnings("unchecked")
+    protected void loadConfig() {
+        InputStream input = null;
+        try {
+            input = new FileInputStream(new File(getInstallLocation() + "/conf/storm.yaml"));
```

Comment:
Again, maybe join the paths the idiomatic way. I think you can do `new File(path1, path2)` in Java.